algorithm-registry.md

```diff
@@ -6,27 +6,51 @@ file **must** be reflected in the `PublicKeyDetails` enumeration in
 [`sigstore_common.proto`] in [sigstore/protobuf-specs].
 
 Sigstore clients aren't required to support all algorithms in this registry,
-and **MAY** support algorithms that aren't in the registry. However,
-compatibility with the Sigstore Public Good Instance requires support
-for at least one of these algorithms.
+and **MAY** support algorithms that aren't in the registry.
+
+See [Spec: Sigstore Public Deployment](./sigstore-public-deployment-spec.md)
+for the algorithms supported by Sigstore's public good instance.
+
+## Algorithm Purposes
+
+The following legend describes the purpose of each algorithm below.
+
+| Symbol | Description | Example |
+|--------|-------------| ------- |
+| 🔑     | End-user signing (ephemeral or long-lived keys) | A user signing with [`cosign`] |
+| 🔗     | TUF metadata signing | [sigstore/root-signing] |
+| 🔏     | Certificate authority materials (CA chains) | [Fulcio] |
+| 🪵     | Certificate transparency log materials (log keys and inclusion proofs) | Fulcio's [CT log] |
+| ⏰     | Timestamp authority materials (TSA chains and signed timestamps) | [sigstore/timestamp-authority] |
+| 📝     | Signature transparency log materials (log keys and inclusion proofs) | [Rekor] |
+| 👀     | Witness keys and signatures | Third-party log witnesses |
```

Comment:
Flagging: I added this legend to make the table below easier to follow, but I could also see it possibly making things more confusing (or too messy with the emojis); curious what others think!